energy-meter-service/src/main/java/com/backend/services/BillGenerationServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.backend.services;
+
+import com.backend.generator.BillGenerator;
+import com.backend.models.request.WeeklyBillRequest;
+import com.backend.models.response.WeeklyBillResponse;
+import org.apache.tomcat.jni.Local;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+public class BillGenerationServiceImpl implements BillGenerationService{
+
+
+    @Override
+    public WeeklyBillResponse getWeeklyBill(WeeklyBillRequest weeklyBillRequest) {
+        BillGenerator billGenerator = new BillGenerator();
```

Comment:
1). Make it as configuration 
2). Rename -> eachDayBillGeneration
3). Autowire it and use instead of 'new BillGenerator()'